src/client-side-encryption/client_encryption.ts

```diff
@@ -956,13 +964,19 @@ export interface ClientEncryptionRewrapManyDataKeyResult {
 /**
  * @public
  * RangeOptions specifies index options for a Queryable Encryption field supporting "rangePreview" queries.
- * min, max, sparsity, and range must match the values set in the encryptedFields of the destination collection.
+ * min, max, sparsity, trimFactor and range must match the values set in the encryptedFields of the destination collection.
  * For double and decimal128, min/max/precision must all be set, or all be unset.
  */
 export interface RangeOptions {
+  /** min is the minimum value for the encrypted index. Required if precision is set. */
   min?: any;
+  /** max is the minimum value for the encrypted index. Required if precision is set. */
   max?: any;
-  sparsity: Long;
+  /** sparsity may be used to tune performance. must be non-negative. When omitted, a default value is used. */
+  sparsity?: Long | bigint | number;
+  /** trimFactor may be used to tune performance. must be non-negative. When omitted, a default value is used. */
+  trimFactor?: Int32 | bigint | number;
```

Comment:
```suggestion
  /** sparsity may be used to tune performance. must be non-negative. When omitted, a default value is used. */
  sparsity?: Long | bigint;
  /** trimFactor may be used to tune performance. must be non-negative. When omitted, a default value is used. */
  trimFactor?: Int32 | number;
```

I'm not sure `bigint` should be permitted for `trimFactor` and `number` for `sparsity`. The difference with those is that `sparsity` must be an `int64` and the only way to accomplish that reliably is with `Long` or `bigint`, you can get an int64 out of a number but it must be very large, seems unlikely that would be a desirable input, but we can leave that one as is if you feel strongly. For `trimFactor` it can be no larger than `int32.max` so only supporting `number` or `Int32` makes sense since bigint will always fail. 

Let's remove `bigint` from `trimFactor` and add tests for these input types. Docs look great, thanks!

Reply:
I'll adjust the types, sure.  But I don't think testing is necessary here - as you've pointed out, this is just standard coercion we're letting users use and if it's not allowed, the server will error.

Reply:
> For trimFactor it can be no larger than int32.max so only supporting number or Int32 makes sense since bigint will always fail.
Do you mean that it always fails because bigints serialize to Longs?  bigints do fit inside the range of int32s.

Reply:
Yes bigints are always serialized as a 64-bit int, so unless we coerce the value to Int32/Number then it will never have the correct BSON type, so I think it would make sense to not list it.

Reply:
Some of it is validated inside libmongocrypt, but I think that's equivalently external and authoritative validation that is better to rely on. I was hoping for some happy path coverage of these types, the tests only use Long and Int32, we can just dupe one of them to use bigint/number too.

Reply:
Okay, I've added tests.